app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCtUtrController.scala

```diff
@@ -0,0 +1,90 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.controllers
+
+import com.google.inject.Inject
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.economiccrimelevyregistration.connectors.EclRegistrationConnector
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.actions.{AuthorisedActionWithEnrolmentCheck, DataRetrievalAction, PublicBetaAction}
+import uk.gov.hmrc.economiccrimelevyregistration.forms.DoYouHaveCtUtrFormProvider
+import uk.gov.hmrc.economiccrimelevyregistration.forms.FormImplicits.FormOps
+import uk.gov.hmrc.economiccrimelevyregistration.models.{CheckMode, Mode, Registration}
+import uk.gov.hmrc.economiccrimelevyregistration.navigation.DoYouHaveCtUtrPageNavigator
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import uk.gov.hmrc.economiccrimelevyregistration.views.html.DoYouHaveCtUtrView
+
+import javax.inject.Singleton
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class DoYouHaveCtUtrController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  authorise: AuthorisedActionWithEnrolmentCheck,
+  getRegistrationData: DataRetrievalAction,
+  formProvider: DoYouHaveCtUtrFormProvider,
+  checkIfPublicBetaIsEnabled: PublicBetaAction,
+  eclRegistrationConnector: EclRegistrationConnector,
+  pageNavigator: DoYouHaveCtUtrPageNavigator,
+  view: DoYouHaveCtUtrView
+)(implicit
+  ec: ExecutionContext
+) extends FrontendBaseController
+    with I18nSupport {
+
+  val form: Form[Boolean] = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData) { implicit request =>
+      Ok(view(form.prepare(request.registration.otherEntityJourneyData.isCtUtrPresent), mode))
+    }
+
+  def onSubmit(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          answer => {
+            val userChangedIsCtutrPresentAnswer = ctutrIsNotPresentAnymore(request.registration, mode, answer)
+            val otherEntityJourneyData          =
+              request.registration.otherEntityJourneyData.copy(
+                isCtUtrPresent = Some(answer),
+                postcode = userChangedIsCtutrPresentAnswer._1,
+                ctUtr = userChangedIsCtutrPresentAnswer._2
```

Comment:
These two lines call the function  `userChangedIsCtutrPresentAnswer` twice!
You can call it only once by doing the following.

`val (postcode, ctutr) = userChangedIsCtutrPresentAnswer`